packages/rendermime/src/widgets.ts

```diff
@@ -63,10 +63,21 @@ export abstract class RenderedCommon extends Widget
    * @param model - The mime model to render.
    *
    * @returns A promise which resolves when rendering is complete.
+   *
+   * #### Notes
+   * If the DOM node for this widget already has content, it is emptied
+   * before rendering. Subclasses that do not want this behavior
+   * (if, for instance, they are using DOM diffing), should override
+   * this method.
    */
   async renderModel(model: IRenderMime.IMimeModel): Promise<void> {
     // TODO compare model against old model for early bail?
 
+    // Empty any existing content in the node from previous renders
+    while (this.node.firstChild) {
+      this.node.removeChild(this.node.firstChild);
+    }
+
```

Comment:
Is it easier to just do `this.node.textContent=''`?

Reply:
huh, https://stackoverflow.com/questions/3955229/remove-all-child-elements-of-a-dom-node-in-javascript indicates your method (or maybe equivalently `this.node.firstChild.remove()`?) may be faster, and possibly more correct if there is, for example, svg content.

Reply:
Yes, I was looking at the same post. I otherwise don't have any reason to chose one method over the other.